dali/operators/image/convolution/gaussian_blur.cc

```diff
@@ -126,36 +121,26 @@ class GaussianBlurOpCpu : public OpImplBase<CPUBackend> {
     int nsamples = input.shape().num_samples();
     for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
       const auto& shape = input[sample_idx].shape();
-      auto elem_volume = volume(shape.begin() + dim_desc_.usable_axes_start, shape.end());
-
-      int seq_elements = 1;
-      int64_t stride = 0;
-      if (dim_desc_.is_sequence()) {
-        seq_elements = volume(shape.begin(), shape.begin() + dim_desc_.usable_axes_start);
-        stride = elem_volume;
-      }
-      for (int elem_idx = 0; elem_idx < seq_elements; elem_idx++) {
-        thread_pool.AddWork(
-            [this, &input, &output, sample_idx, elem_idx, stride](int thread_id) {
-              auto gaussian_windows = windows_[sample_idx].GetWindows();
-              auto elem_shape = input[sample_idx].shape().template last<ndim>();
-              auto in_view = TensorView<StorageCPU, const In, ndim>{
-                  input[sample_idx].template data<In>() + stride * elem_idx, elem_shape};
-              auto out_view = TensorView<StorageCPU, Out, ndim>{
-                  output[sample_idx].template mutable_data<Out>() + stride * elem_idx, elem_shape};
-              // I need a context for that particular run (or rather matching the thread &
-              // scratchpad)
-              auto ctx = ctx_;
-              kmgr_.Run<Kernel>(sample_idx, ctx, out_view, in_view, gaussian_windows);
-            }, elem_volume);
-      }
+      auto elem_volume = volume(shape);
+      thread_pool.AddWork(
+          [this, &input, &output, shape, sample_idx](int thread_id) {
+            auto gaussian_windows = windows_[sample_idx].GetWindows();
+            auto in_view = TensorView<StorageCPU, const In, ndim>{
+                input[sample_idx].template data<In>(), shape};
+            auto out_view = TensorView<StorageCPU, Out, ndim>{
+                output[sample_idx].template mutable_data<Out>(), shape};
+            // I need a context for that particular run (or rather matching the thread &
+            // scratchpad)
+            auto ctx = ctx_;
```

Comment:
Use DynamicScratchpad and a local context.

Reply:
I guess it may be a minor change code-wise but given the extent of this PR I'd like to make it a separate PR.

Reply:
OK. It's not that important for a purely CPU op anyway.